### ISSUE TITLE
Uninitialised Classifier or invalid context

### ISSUE THREAD

**Issue by aravindsasidharan** [tensorflow/tensorflow#15751]
I was able to run the tensorflowlite demo app with no error. But it is not detecting the objects as it is supposed to. I have followed the instructions to be followed  in android studio in readme.
![Uploading Screenshot_20171231-151052.png…]()

**Comment by bignamehyp** [comment]
I apologize but I am having a hard time understanding what the problem is, where the problem is, and what version it affects. Please resubmit and pay attention to the issue template (https://github.com/tensorflow/tensorflow/issues/new) . Please provide all the information it asks. Thank you.

**Comment by AtomicSpider** [comment]
I'm having the same issue. 

In "Camera2BasicFragment.java -> classifyFrame()"

    if (classifier == null || getActivity() == null || cameraDevice == null) {
        showToast("Uninitialized Classifier or invalid context.");
        return;
    }

this function throws the error. I checked to see exactly what was wrong and apparently, classifier was null.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by RakeshChandran** [comment]
Yes..I am having the same issue. @tensorflowbutler. What is the resolution ?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by LinghaoYu** [comment]
I have the same issue also.

**Comment by ahdidou-mohamed** [comment]
Hi there , 

Did you copy the tflite file in the assets folder ? 

I had the same issue , but got the demo project working by copying this file  **_mobilenet_quant_v1_224.tflite_** in the asset folder.

**Comment by zob** [comment]
3 ways to correct this error.
1: You can add a labels file in assets called labels.txt
2: rename the given file in assets called labels_mobilenet_quant_v1_224.txt to labels.txt
3: change the returned value in `getLabelPath()` in ImageClassifierQuantizedMobileNet.java to `return "labels_mobilenet_quant_v1_224.txt";`

**Comment by HandsomeL** [comment]
I get the same error!
It caused by MISSING .tflite/.
fitstly  you need download the files at:[https://storage.googleapis.com/download.tensorflow.org/models/tflite/mobilenet_v1_224_android_quant_2017_11_08.zip](https://storage.googleapis.com/download.tensorflow.org/models/tflite/mobilenet_v1_224_android_quant_2017_11_08.zip)

**Comment by rupvm** [comment]
I have the same issue also.

mobilenet_quant_v1_224.tflite missing copying this file in the assets folder.

issue solved. !!

thanks @ahdidou-mohamed @HandsomeL

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!

**Comment by zhyj3038** [comment]
I follow the "image retrain" retrain my own data. if I down load official .pb and convert to .lite , I can run on android device successfully.  but i I train my own data using mobile v2, error occurs "Uninitialised Classifier or invalid context"

**Comment by monircse** [comment]
@tensorflowbutler I'm facing the same issue. I have renamed my labels.txt and model.tflite as per my codes. Still getting the same error. Anybody please to help?
![image](https://user-images.githubusercontent.com/6847030/63915401-d5d54480-ca57-11e9-9b95-84c5e6e420eb.png)

**Comment by Sam-songchen** [comment]
how to solve the problem,Does anyone have a solution

**Comment by Nadav89** [comment]
Having the same problem, was anyone able to fix it?

**Comment by mimikuo365** [comment]
Hi there! I worked on the problem for several days and would like to share how I solved it.
I use `adb logcat > log.txt` to get the error code. 
Program crashes with `FATAL EXCEPTION: CameraBackground` and `java.lang.IllegalArgumentException: Cannot copy to a TensorFlowLite tensor (image) with 150528 bytes from a Java Buffer with 602112 bytes.`. 
After searching on Stack Overflow, I solved the problem with [this link](https://stackoverflow.com/q/67419320).
Hope this is helpful!
